src/stores/contactmoment/index.ts

```diff
@@ -1,102 +1,151 @@
   },
   getters: {
-    klant: (state): Klant | undefined =>
-      state.klanten.filter((x) => x.shouldStore).map((x) => x.klant)[0],
+    klant(state): Klant | undefined {
+      return state.huidigeVraag.klanten
+        ?.filter((x) => x.shouldStore)
+        ?.map((x) => x.klant)?.[0];
+    },
+    huidigeVraagIndex: (state) => state.vragen.indexOf(state.huidigeVraag),
   },
   actions: {
     start() {
+      if (this.contactmomentLoopt) return;
       this.contactmomentLoopt = true;
     },
+    startNieuweVraag() {
+      if (!this.contactmomentLoopt) return;
+      const nieuweVraag = initVraag();
+      if (this.huidigeVraag.klanten) {
+        nieuweVraag.klanten = this.huidigeVraag.klanten.map(
+          (klantKoppeling) => ({
+            ...klantKoppeling,
+          })
+        );
+      }
+      this.vragen.push(nieuweVraag);
+      this.huidigeVraag = nieuweVraag;
+    },
     stop() {
       this.$reset();
       // Temporary. When we implement multiple running contactmomenten, each will have it's own state
       resetAllState();
     },
-    addZaak(zaak: Zaak) {
-      const contactmomentZaak = zaak as ContactmomentZaak;
-      const index = this.zaken.findIndex((element) => element.id === zaak.id);
-      if (index === -1) {
-        //als de zaak nog niet gekoppeld was aan het contact moment dan voegen we hem eerst toe
-        this.zaken.push(contactmomentZaak);
-        contactmomentZaak.shouldStore = true;
-      } else {
-        const existingZaak = this.zaken[index];
-        existingZaak.shouldStore = true;
+    addZaak(zaak: Zaak, vraag: Vraag) {
+      const match = vraag.zaken.find((element) => element.zaak.id === zaak.id);
+
+      if (match) {
+        match.zaak = zaak;
+        match.shouldStore = true;
+        return;
       }
+
+      //als de zaak nog niet gekoppeld was aan het contact moment dan voegen we hem eerst toe
+      vraag.zaken.push({
+        zaak,
+        shouldStore: true,
+      });
     },
-    toggleZaak(zaak: Zaak) {
-      const contactmomentZaak = zaak as ContactmomentZaak;
-      const index = this.zaken.findIndex((element) => element.id === zaak.id);
-      if (index === -1) {
-        //als de zaak nog niet gekoppeld was aan het contact moment dan voegen we hem eerst toe
-        this.zaken.push(contactmomentZaak);
-        contactmomentZaak.shouldStore = true;
-        return true;
-      } else {
-        const existingZaak = this.zaken[index];
-        //toggle of hij wel niet opgeslagen moet worden bij het contactmoment
-        existingZaak.shouldStore = !existingZaak.shouldStore;
-        return existingZaak.shouldStore;
+    toggleZaak(zaak: Zaak, vraag: Vraag) {
+      const match = vraag.zaken.find((element) => element.zaak.id === zaak.id);
+
+      if (match) {
+        match.zaak = zaak;
+        match.shouldStore = !match.shouldStore;
+        return;
       }
+
+      vraag.zaken.push({
+        zaak,
+        shouldStore: true,
+      });
```

Comment:
This toggle and `addZaak` seem to be identical, apart from the `shouldStore` part. Might be worthwhile to combine them.

Reply:
Agreed. This was already the case, but I refactored this code slightly already, so might as well fix this too.